src/map/latent_effect_container.cpp

```diff
@@ -1674,7 +1675,23 @@ void CLatentEffectContainer::CheckLatentsZone()
             }
             break;
         case LATENT_WEATHER_ELEMENT:
-            if (zoneutils::GetWeatherElement(zoneutils::GetZone(m_POwner->getZone())->GetWeather()) == m_LatentEffectList.at(i)->GetConditionsValue())
+            // In the default case we need to pull weather from zone/player
+            if (weather == MAX_WEATHER_ID)
```

Comment:
regardless of what you do, if the value for the weather is *ever* the same as the value of MAX_WEATHER_ID your map server is going to crash on a debug break. You should never do either setting it to that or checking for it being that, because it should never be that value.